library/ssl_cli.c

```diff
@@ -3351,12 +3356,12 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
                                         ssl->handshake->premaster,
                                         sizeof( ssl->handshake->premaster ),
                                         &ssl->handshake->pmslen );
-        if( status != PSA_SUCCESS )
-            return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
 
-        status = psa_destroy_key( handshake->ecdh_psa_privkey );
-        if( status != PSA_SUCCESS )
+        destruction_status = psa_destroy_key( handshake->ecdh_psa_privkey );
+
+        if( status != PSA_SUCCESS || destruction_status != PSA_SUCCESS )
             return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
+
         handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT;
```

Comment:
Perhaps `handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT` should be moved right after `psa_destroy_key`.